hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1354,6 +1389,9 @@ int BleObject::Observer::stopScanning() {
     if (give) {
         os_semaphore_give(scanSemaphore_, false);
     }
+    if (os_timer_is_active(scanGuardTimer_, nullptr)) {
```

Comment:
Stop before unblocking the thread that requested scanning?

Reply:
The timer is stopped on entering the function.

Reply:
Can you point to where that happens?

Reply:
I meant I just moved it up in the same function ahead of giving the scan semaphore.